src/main/scala/com/moesif/filter/MoesifApiFilter.scala

```diff
@@ -265,7 +293,14 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
       lastSendTime = System.currentTimeMillis()
 
       while(eventModelCache.nonEmpty){
-        val sendingEvents: Seq[EventModel] = eventModelCache.take(batchSize)
+        var sendingEvents: Seq[EventModel] = eventModelCache.take(batchSize)
+        var sendingEventsSize = APIHelper.serialize(sendingEvents).getBytes.length
+        while (sendingEventsSize >= maxBatchSize) {
+          // if the batch size exceeds maxBatchSize, reduce the batch size by half
+          val reducedSize = sendingEvents.size * 0.9 // reduce the number of events by 10%
+          sendingEvents = eventModelCache.take(reducedSize.toInt)
```

Comment:
If one event is larger than maxBatchSize, this will reduce to zero, take no events, and turn the outer while loop into an infinite loop.  This needs some kind of checks around, but I'm not sure if that should be on the config load size with a minimum allowed maxBatchSize that's bigger than out max event size in collector, customer docs, or a runtime error, but something